### PR TITLE
[DataPipe] Fix OnDiskCacheHolder to list all files for decompressing operations

### PR DESCRIPTION
**Pull request by ejguan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #202 Revamp librispeech dataset
* **#203 [DataPipe] Fix OnDiskCacheHolder to list all files for decompressing operations**
* #201 Implement DataPipe to convert Iter to Map

Add `FileLister` to make sure `OnDiskCacheHolder` can list all of files after any 1-to-N operations like decompression.

Differential Revision: [D34085743](https://our.internmc.facebook.com/intern/diff/D34085743)

### CODE REVIEW

File: test/test_remote_io.py [hunk 1]
```diff
@@ -137,6 +137,30 @@ def _read_and_decode(x):
             self.assertTrue(os.path.exists(expected_csv_path))
             self.assertEqual(expected_csv_path, csv_path)
 
+        # Cache decompressed archive but only check root directory
```
Comment (ejguan) [inline test/test_remote_io.py:140]: The following test won't pass with previous implementation.

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34085743).

**Review by NivekT — APPROVED** [review]
LGTM!

**Review by VitalyFedyunin — APPROVED** [review]

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34085743).

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34085743).

**Comment by ejguan** [comment]
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34085743).

**Pull request closed without merging**